design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,73 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new field in BareMetalHost so that it is possible to attach a generic
+non-bootable iso image via Ironic, after the host has been provisioned.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example :
+
+As a config data disk to do some post provisioning configuration for the
+system.
+
+For troubleshooting of the node.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new optional section in the BareMetalHost spec containing details of the
+non-bootable iso image, example :
+
+```yaml
+  spec:
+    dataImage:
+      url: http://1.2.3.4/image.iso
+      username: metal
+      password: password
+      insecure: false
+```
+
+Note that in this case, the iso image will be attached after the node
+has been provisioned or ExternallyProvisioned.
+
+### Ironic Proposal
+
+The [proposal](https://bugs.launchpad.net/ironic/+bug/2033288) for exposing this functionaility in Ironic has been
+accepted by the upstream Ironic community.
+
+## Design Details
+
+Add an optional section `DataImage` to `BareMetalHostSpec`, with sub-fields
+`Url`, `Username`, `Password`, `Insecure`.
```

Comment:
This information is clear from the previous sections. What I'd like to see here are some details on how the new operation will interact with other BMH states. E.g. when I create a BMH with both `image` and `dataImage` set, at which point is the data image connected?

If I create an externally provisioned host with `online: true` and `dataImage`, we want to make sure we connect the image *before* we try to power on the host.

And so on.